### PR TITLE
SDK-10939 - add onChangeTrackedCodes callback

### PR DESCRIPTION
**Pull request by petradonka**

### CODE REVIEW

**Comment by TomekJurkowski** [comment]
would you like to add documentation for the new `onChangeTrackedCodes` method in Scandit.java doc file?
Alternatively, we can merge this as is and I'll add the docs with the Android MR

File: components/MatrixScanSession.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 export class MatrixScanSession {
 
-	constructor(newlyTrackedCodes) {
+	constructor(newlyTrackedCodes, allTrackedCodes) {
```
[the remark this answers is not in the document]
Comment (petradonka) [reply components/MatrixScanSession.js]: tabs? 😱 of course i can replace them 😄

[the remark this answers is not in the document]
Comment (petradonka) [reply components/MatrixScanSession.js]: done ✅

**Comment by petradonka** [comment]
@TomekJurkowski I added documentation, also for some other missing things, could you check and make sure that at least they look like they'll show up fine after generation? 😄

File: ios/ScanditBarcodeScanner/SCNBarcodePicker.m [hunk 2]
```diff
@@ -306,27 +348,43 @@ - (void)barcodePicker:(nonnull SBSBarcodePicker *)barcodePicker
...
-                SBSTrackedCode *code = session.trackedCodes[identifier];
-                [session rejectTrackedCode:code];
-            }
-            self.idsToVisuallyReject = nil;
+        dispatch_semaphore_wait(self.didFinishOnRecognizeNewCodesSemaphore, DISPATCH_TIME_FOREVER);
+        [self handleFinishingSemaphore:session];
+    }
+}
+
+- (void) handleFinishingSemaphore:(SBSScanSession *)session {
```
Comment (lucatorella) [inline ios/ScanditBarcodeScanner/SCNBarcodePicker.m]: `(void) handleFinishingSemaphore` -> `(void)handleFinishingSemaphore`

Comment (petradonka) [reply ios/ScanditBarcodeScanner/SCNBarcodePicker.m]: ✅

File: docs/Scandit.java [hunk 3]
```diff
@@ -1196,6 +1246,81 @@
...
+    * {@link ScanSettings.codeRejectionEnabled code rejection} to true.
+    *
+    * Rejected codes will be added to {@link allRecognizedCodes} like all other codes.
+    *
+    * Note that you should only pass codes returned by {@link newlyRecognizedCodes}
+    * as passing any other code will have no effect. Additionally, you should only
+    * calls this method from the scan callback.
+    *
+    * @param code The code to reject
+    *
```
Comment (TomekJurkowski) [inline docs/Scandit.java]: \n\n ;)

File: docs/Scandit.java [hunk 4]
```diff
@@ -1196,6 +1246,81 @@
...
+    *
+    */
+    public void rejectCode(Barcode code);
+  }
+  public class MatrixScanSession {
+
+    /**
+    * List of barcodes that are newly tracked in the last frame.
+    *
```
Comment (TomekJurkowski) [inline docs/Scandit.java]: \n\n

File: docs/Scandit.java [hunk 5]
```diff
@@ -1196,6 +1246,81 @@
...
+    * This is useful for briefly pausing the barcode recognition to show the
+    * recognized code in an overlay and then resume the scan process to scan
+    * more codes.
+    * <p>
+    * When only scanning one code and then returning to another part of the
+    * application, it is recommended to call {@link stopScanning()} instead.
+    * <p>
+    *
+    * @see BarcodePicker.resumeScanning()
+    *
```
Comment (TomekJurkowski) [inline docs/Scandit.java]: \n\n

File: docs/Scandit.java [hunk 6]
```diff
@@ -1196,6 +1246,81 @@
...
+    public void pauseScanning();
+
+    /**
+    * Immediately stops the scanning and clears the scan session
+    * <p>
+    * Calling stop will release the camera, so that other applications can use
+    * it.
+    *
+    * @see BarcodePicker.stopScanning(), pauseScanning()
+    *
```
Comment (TomekJurkowski) [inline docs/Scandit.java]: \n\n

File: docs/Scandit.java (hunk 3 above)
Comment (petradonka) [reply docs/Scandit.java]: ✅

File: docs/Scandit.java (hunk 4 above)
Comment (petradonka) [reply docs/Scandit.java]: ✅

File: docs/Scandit.java (hunk 5 above)
Comment (petradonka) [reply docs/Scandit.java]: ✅

File: docs/Scandit.java (hunk 6 above)
Comment (petradonka) [reply docs/Scandit.java]: ✅

**Pull request merged**